turn_on_led/turn_on_led/turn_on_led.asm

```diff
@@ -0,0 +1,659 @@
+;When to the column on the display is provided low, and to row is provided high then the led is lit
+;For the column pattern 0b11111110 the first led on the left will be lit
+;For the row pattern 0b00000001 the top row is lit
+
+;define names for registers
+.def ballCol = r17
+.def ballRow = r18
+.def paddleCol = r19
+.def vectorX = r23
+.def vectorY = r24
+.def blockRow1 = r26
+.def blockRow2 = r27
+.def blockRow3 = r28
+.def blockRow4 = r29
+.def blockCounter = r10
+
+
+;free registers: r16, r30, r31
+
+ldi r16, 0b11111111     ; writing bit pattern 1000 0000 to register 16
+out ddra, r16           ; setup data direction for port A - output
+out ddrc, r16			; setup data direction for port C - output
+
+;prepare game registers
+ldi ballCol, 0b11101111 ; ball column - 0 means that column is displayed
+ldi ballRow, 0b01000000 ; ball row - 1 means that ball is in that raw
+ldi vectorX, 0b00000001 ; direction of vector (100 left / 001 right)
+ldi vectorY, 0b00000010 ; direction of vector (10 up / 01 down)
+
+;initialize block pattern
+ldi blockRow1, 0b10101010
+ldi blockRow2, 0b01010101
+ldi blockRow3, 0b10101010
+ldi blockRow4, 0b01010101
+
+;initialize blocks to hit counter. When counter = 0 then the game is over and player has won
+ldi r16, 16
+mov blockCounter, r16
+
+ldi paddleCol, 0b11000111 ; paddle column (0 - led on, 1 - led off) each bit corresponds to one column 
+
+ldi		r20, 0xff		; left button
+out		ddrg, r20	    ; set port G to input
+
+ldi		r20, 0xff		; right button
+out		ddrb, r20	    ; set port B to input
+
+
+game_start:  ; loading start position, the game will start when both buttons are pressed 
+	call start_position
+	in r20, PING		;read left button
+	cpi r20, 0			;if r20 != 0 then button is pressed
+	breq game_start
+	in r20, PINB		;read right button
+	cpi r20, 0			;if r20 != 0 then button is pressed
+	breq game_start		;go back in loop
+
+main:
+	ldi r25, 12 ; number of display loop iteration before game logic
+	
+display:
+	;section responsible for displaying the game status on the screen
+	;each row is displayed separately in 10ms periods. Then the procedure repeats
+	call row_one
+	call delay_10ms
+
+	call row_two
+	call delay_10ms
+
+	call row_three
+	call delay_10ms
+
+	call row_four
+	call delay_10ms
+	
+	call ball
+	call delay_10ms
+
+	call paddle
+	call delay_10ms
+	;displaying all elements on the screen takes approx. 60ms
+
+	
+	;when the screen is displayed 12 times, 
+	;so 12*60ms = 720ms passed the game logic is checked
+	dec r25
+	cpi r25, 0
+	breq game
+	rjmp display
+
+
+game:
+	call button					; read buttons input
+	
+	call check_ball_left_wall	
+	call check_ball_right_wall
+
+	call check_ball_hit_paddle
+
+	call check_ball_hit_block
+
+	call check_ball_top
+
+	;move the ball according to vector 
+	call movementX				; move ball in X axis
+	call movementY				; move ball in Y axis
+
+	call check_win
+	call check_game_over
+	rjmp main
```

Comment:
I like your clear structure. It enables me to get an quick overview of the entire application.